### PR TITLE
Added invalid domain check, invalid core check, invalid chip check

### PR DESCRIPTION
**Pull request by Krishan-Saraswat**
[root@perf]# avocado run --max-parallel-tasks=1 perf_24x7_hardware_counters.py
JOB ID     : 2ca053ef8038d83ed373a9a289f6911b384d68fa
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2023-09-11T13.12-2ca053e/job.log
 (01/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_display_domain_indices_in_sysfs: STARTED
 (01/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_display_domain_indices_in_sysfs: PASS (1.63 s)
 (02/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_event_phys_core_param: STARTED
 (02/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_event_phys_core_param: CANCEL: HPM_0THRD_NON_IDLE_CCYC__PHYS_CORE not found (36.69 s)
 (03/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_event_wo_domain_param: STARTED
 (03/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_event_wo_domain_param: PASS (1.70 s)
 (04/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_check_all_domains: STARTED
 (04/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_check_all_domains: PASS (14.49 s)
 (05/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_check_invalid_domains: STARTED
 (05/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_check_invalid_domains: FAIL: perf unable to recognize invalid domain (1.74 s)
 (06/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_check_invalid_core: STARTED
 (06/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_check_invalid_core: PASS (1.70 s)
 (07/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_event_w_chip_param: STARTED
 (07/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_event_w_chip_param: PASS (1.63 s)
 (08/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_event_wo_chip_param: STARTED
 (08/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_event_wo_chip_param: PASS (1.70 s)
 (09/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_check_valid_chip: STARTED
 (09/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_check_valid_chip: PASS (2.20 s)
 (10/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_check_invalid_chip: STARTED
 (10/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_check_invalid_chip: PASS (1.77 s)
 (11/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_domain_chip_offset: STARTED
 (11/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_domain_chip_offset: PASS (102.64 s)
 (12/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_event_helper_phys_core: STARTED
 (12/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_event_helper_phys_core: PASS (1.68 s)
 (13/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_event_helper_vcpu: STARTED
 (13/13) perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_event_helper_vcpu: PASS (1.69 s)
RESULTS    : PASS 11 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2023-09-11T13.12-2ca053e/results.html
JOB TIME   : 270.13 s

Test summary:
perf_24x7_hardware_counters.py:EliminateDomainSuffix.test_check_invalid_domains: FAIL

[job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/12579284/job.log)

### CODE REVIEW

**Review by disgoel — APPROVED** [review]
LGTM

File: perf/perf_24x7_hardware_counters.py [hunk 1]
```diff
@@ -90,9 +90,18 @@ def setUp(self):
...
+        output = process.run("lscpu")
+        for line in output.stdout.decode("utf-8").split('\n'):
+            if 'Core(s) per socket:' in line:
+                self.cores = int(line.split(':')[1].strip())
+            if 'Physical sockets:' in line:
+                self.physical_sockets = int(line.split(':')[1].strip())
+            if 'Physical chips:' in line:
+                self.physical_chips = int(line.split(':')[1].strip())
+        # chip = physical socket x physical chip
+        self.chips = int(self.physical_sockets*self.physical_chips)
```
Comment (PraveenPenguin) [inline perf/perf_24x7_hardware_counters.py:104]: can we push this as part of avocado utils?

File: perf/perf_24x7_hardware_counters.py [hunk 2]
```diff
@@ -131,26 +140,39 @@ def test_event_wo_domain_param(self):
...
+                if "Performance counter stats for" not in result1.stderr.decode("utf-8"):
+                    self.fail('perf unable to recognize domain name in'
+                              ' param=value format for all domains')
+                else:
+                    self.log.info('perf recognized domain name in param=value'
+                                  ' format for all 6 domains')
+
+    def test_check_invalid_domains(self):
+        # check invalid domains
+        invalid_domains = [0, 7, 16]
```
[the remark this answers is not in the document]
Comment (Krishan-Saraswat) [reply perf/perf_24x7_hardware_counters.py:160]: yes, these are constant values for invalid domains.

File: perf/perf_24x7_hardware_counters.py [hunk 3]
```diff
@@ -131,26 +140,39 @@ def test_event_wo_domain_param(self):
...
+            if result.exit_status == 0 and b"not supported" in result.stderr:
+                self.log.info("perf recognized domain as invalid domains")
+            elif domain == 16 and result.exit_status != 0:
+                self.log.info("perf recognized domain as invalid domain")
             else:
-                self.log.info('perf recognized domain name in param=value'
-                              ' format for all 6 domains')
+                self.fail("perf unable to recognize invalid domain")
+
+    def test_check_invalid_core(self):
```
Comment (Naresh-ibm) [inline perf/perf_24x7_hardware_counters.py:170]: Add the Docstring

File: perf/perf_24x7_hardware_counters.py [hunk 4]
```diff
@@ -180,13 +202,29 @@ def test_event_wo_chip_param(self):
...
             self.log.info('perf detected chip parameter missing')
-        if self.rev == '004b' or self.rev == '004e':
-            cmd = "hv_24x7/PM_PB_CYC,domain=1,chip=1/ /bin/true"
-        if self.rev == '0080':
-            cmd = "hv_24x7/PM_PHB0_0_CYC,domain=1,chip=1/ /bin/true"
-        output_chip = self.event_stat1(cmd)
-        if "Performance counter stats for" not in output_chip.stderr.decode("utf-8"):
-            self.fail('performance counter stats for missing')
+
+    def test_check_valid_chip(self):
```
Comment (Naresh-ibm) [inline perf/perf_24x7_hardware_counters.py:206]: dockstring

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

**Comment by disgoel** [comment]
@PraveenPenguin @Naresh-ibm can you please review and close

**Review by PraveenPenguin — APPROVED** [review]

**Comment by PraveenPenguin** [comment]
@Naresh-ibm can you please review this PR

**Review by Naresh-ibm — APPROVED** [review]
LGTM

**Comment by disgoel** [comment]
> Hi Krishan,any reason on removing code to print " error msg to enable permission tp collect stats?" I think its better to have that code as many time, there is wrong alarm because of this option not enabled.

There were two checks for the same thing, so he removed the old check which was no more valid error message. Below check is still there.
```
        if "operations is limited" in result_perf.stderr.decode("utf-8"):
            self.cancel("Please enable lpar to allow collecting"
                        " the 24x7 counters info")
```

File: perf/perf_24x7_hardware_counters.py [hunk 5]
```diff
@@ -131,26 +132,41 @@ def test_event_wo_domain_param(self):
...
-                self.log.info('perf recognized domain name in param=value'
-                              ' format for all 6 domains')
+                self.fail("perf unable to recognize invalid domain")
+
+    def test_check_invalid_core(self):
+        """
+        supported core value 0-65535
+        check invalid core value
+        """
+        result = self.event_stat(',domain=3,core=65536/ sleep 1')
```
[the remark this answers is not in the document]
Comment (disgoel) [reply perf/perf_24x7_hardware_counters.py:167]: addressed this in PR https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2808

**Pull request merged**